packages/cli/src/databases/migrations/sqlite/1728396464278-AddMissingPrimaryKeyOnAnnotationTagMapping.ts

```diff
@@ -0,0 +1,74 @@
+import { ApplicationError } from 'n8n-workflow';
+
+import type { IrreversibleMigration, MigrationContext } from '@/databases/types';
+
+const annotationsTableName = 'execution_annotations';
+const annotationTagsTableName = 'annotation_tag_entity';
+const annotationTagMappingsTableName = 'execution_annotation_tags';
+
+export class AddMissingPrimaryKeyOnAnnotationTagMapping1728396464278
+	implements IrreversibleMigration
+{
+	async up({
+		queryRunner,
+		tablePrefix,
+		schemaBuilder: { createTable, column, dropIndex },
+	}: MigrationContext) {
+		// Check if the primary key already exists
+		const table = await queryRunner.getTable(`${tablePrefix}execution_annotation_tags`);
+
+		if (!table) {
+			throw new ApplicationError('execution_annotation_tags table not found');
+		}
+
+		const hasPrimaryKey = table.primaryColumns.length > 0;
+
+		// Do nothing if the primary key already exists
+		if (hasPrimaryKey) {
+			return;
+		}
+
+		// SQLite doesn't support adding a primary key to an existing table
+		// So we have to do the following steps:
+
+		// 1. Rename the existing table
+		await queryRunner.query(
+			`ALTER TABLE ${tablePrefix}${annotationTagMappingsTableName} RENAME TO ${tablePrefix}${annotationTagMappingsTableName}_tmp;`,
+		);
+
+		// 1.1 Drop the existing indices
+		await dropIndex(`${annotationTagMappingsTableName}_tmp`, ['tagId'], {
+			customIndexName: 'IDX_a3697779b366e131b2bbdae297',
+		});
+		await dropIndex(`${annotationTagMappingsTableName}_tmp`, ['annotationId'], {
+			customIndexName: 'IDX_c1519757391996eb06064f0e7c',
+		});
```

Comment:
Do we need to drop these if the tmp table gets dropped at the end? Does that also drop these indexes?

Reply:
Yes, otherwise it fails on creating new table, as indexes with these names are already exist.

Reply:
You could give the new indexes custom names to prevent this as well. 
I had to do this here as well:

https://github.com/n8n-io/n8n/blob/2e6b03b2cb471aefa8104b7b80cf12e64f16e4fb/packages/cli/src/databases/migrations/common/1720101653148-AddConstraintToExecutionMetadata.ts#L83-L87